src/components/richTextEditor/MenuBar.tsx

```diff
@@ -0,0 +1,188 @@
+      isActive: () => editor.isActive('heading', { level: 1 }),
+    },
+    {
+      icon: Heading2,
+      title: 'Heading 2',
+      action: () => editor.chain().focus().toggleHeading({ level: 2 }).run(),
+      isActive: () => editor.isActive('heading', { level: 2 }),
+    },
+    {
+      icon: Left,
+      title: 'Left',
+      action: () => editor.chain().focus().setTextAlign('left').run(),
+      isActive: () => editor.isActive({ textAlign: 'left' }),
+    },
+    {
+      icon: Center,
+      title: 'Center',
+      action: () => editor.chain().focus().setTextAlign('center').run(),
+      isActive: () => editor.isActive({ textAlign: 'center' }),
+    },
+    {
+      icon: Right,
+      title: 'Right',
+      action: () => editor.chain().focus().setTextAlign('right').run(),
+      isActive: () => editor.isActive({ textAlign: 'right' }),
+    },
+    
+    {
+      icon: Paragraph,
+      title: 'Paragraph',
+      action: () => editor.chain().focus().setParagraph().run(),
+      isActive: () => editor.isActive('paragraph'),
+    },
+    {
+      icon: BulletList,
+      title: 'Bullet List',
+      action: () => editor.chain().focus().toggleBulletList().run(),
+      isActive: () => editor.isActive('bulletList'),
+    },
+    {
+      icon: OrderedList,
+      title: 'Ordered List',
+      action: () => editor.chain().focus().toggleOrderedList().run(),
+      isActive: () => editor.isActive('orderedList'),
+    },
+    {
+      icon: TaskList,
+      title: 'Task List',
+      action: () => editor.chain().focus().toggleTaskList().run(),
+      isActive: () => editor.isActive('taskList'),
+    },
+    {
+      icon: CodeBlock,
+      title: 'Code Block',
+      action: () => editor.chain().focus().toggleCodeBlock().run(),
+      isActive: () => editor.isActive('codeBlock'),
+    },
+   
+    {
+      icon: Blockquote,
+      title: 'Blockquote',
+      action: () => editor.chain().focus().toggleBlockquote().run(),
+      isActive: () => editor.isActive('blockquote'),
+    },
+    {
+      icon: HorizontalRule,
+      title: 'Horizontal Rule',
+      action: () => editor.chain().focus().setHorizontalRule().run(),
+     
+     
+    },
+   
+    {
+      icon: HardBreak,
+      title: 'Hard Break',
+      action: () => editor.chain().focus().setHardBreak().run(),
+     
+    },
+    {
+      icon: ClearFormat,
+      title: 'Clear Format',
+      action: () => editor.chain().focus().clearNodes().unsetAllMarks()
+        .run(),
+        
+    },
+  
+    {
+      icon: Undo,
+      title: 'Undo',
+      action: ()  => editor.chain().focus().undo().run(),
+      
+    },
+    {
+      icon: Redo,
+      title: 'Redo',
+      action: () => editor.chain().focus().redo().run(),
+      
+    },
+  ]
+  
+  return (
+    <div className="editor__header">
+      {items.map((item, index) => (
+          <MenuItem key={index} {...item as MenuItemProps} isActive={item.isActive } />
+      ))}
+    </div>
+  )
+}
+
+export default MenuBar;
```

Comment:
it is better for the file to have one empty line at the end

you make your research to learn more about that